discovery-infra/test_infra/utils.py

```diff
@@ -612,18 +612,26 @@ def file_lock_context(filepath='/tmp/discovery-infra.lock', timeout=300):
     finally:
         lock.release()
 
+def _get_hosts_from_network(net):
+    desc = md.parseString(net.XMLDesc())
+    try:
+        hosts = desc.getElementsByTagName("network")[0].getElementsByTagName("ip")[0].getElementsByTagName("dhcp")[0].getElementsByTagName("host")
+        return list(map(lambda host: {"mac": host.getAttribute("mac"), "ipaddr": host.getAttribute("ip"), "hostname": host.getAttribute("name")}, hosts))
+    except IndexError:
+        return []
+
+def _merge(leases, hosts):
+    lips = [l["ipaddr"] for l in leases]
+    ret = leases + [h for h in hosts if h["ipaddr"] not in lips]
```

Comment:
Shouldn't it check by hostname?

Reply:
Hostname does not always appear as part of the lease

Reply:
IP yes and hostname no? wouldn't it break other code depending on that?
https://github.com/openshift/assisted-test-infra/blob/e211a3fff26aa3d3d2f62a3ba7e758e3b1af3b82/discovery-infra/test_infra/utils.py#L96

Reply:
Hostname is not mandatory in lease.  In IPv6 we don't have hostnames